GDJS/Runtime/howler-sound-manager/howler-sound-manager.ts

```diff
@@ -232,84 +425,184 @@ namespace gdjs {
      */
     private _storeSoundInArray(arr: Array<HowlerSound>, sound): any {
       //Try to recycle an old sound.
-      for (let i = 0, len = arr.length; i < len; ++i) {
-        if (arr[i] !== null && arr[i].canBeDestroyed()) {
+      for (var i = 0, len = arr.length; i < len; ++i) {
+        if (arr[i] !== null && arr[i].stopped()) {
           arr[i] = sound;
           return sound;
         }
       }
+
       arr.push(sound);
       return sound;
     }
 
-    playSound(soundName, loop, volume, pitch) {
+    /**
+     * Creates a new gdjs.HowlerSound using preloaded/cached Howl instances.
+     * @param soundName The name of the file or resource to play.
+     * @param isMusic True if a music, false if a sound.
+     */
+    createHowlerSound(soundName: string, isMusic: boolean): HowlerSound {
       const soundFile = this._getFileFromSoundName(soundName);
-      const sound = new gdjs.HowlerSound({
+      const cacheContainer = isMusic ? this._loadedMusics : this._loadedSounds;
+
+      if (!cacheContainer.hasOwnProperty(soundFile)) {
+        cacheContainer[soundFile] = new Howl({
+          ...HowlParameters,
+          src: [soundFile],
+          html5: isMusic,
+        });
+      }
+      return new gdjs.HowlerSound(cacheContainer[soundFile]);
+    }
+
+    /**
+     * Preloads a sound or a music in memory.
+     * @param soundName The name of the file or resource to preload.
+     * @param isMusic True if a music, false if a sound.
+     */
+    loadAudio(soundName: string, isMusic: boolean) {
+      const soundFile = this._getFileFromSoundName(soundName);
+      const cacheContainer = isMusic ? this._loadedMusics : this._loadedSounds;
+
+      // Do not reload if it is already loaded.
+      if (cacheContainer.hasOwnProperty(soundFile)) return;
+
+      cacheContainer[soundFile] = new Howl({
+        ...HowlParameters,
```

Comment:
Is it ok to use the spread operator or is it too modern javascript?

Reply:
Good question, it might be a bit too early, let's avoid it to try to ensure Android 5 compatibility, even if the updatable webview in Android 6 means that we should not get outdated Chrome at some point.